README.md

```diff
@@ -0,0 +1,222 @@
+# Kernel Settings Role
+
+This role is used to modify kernel settings.  It uses `tuned` for its
+implementation, so it is able to manage all of the settings managed by `tuned`
+such as `sysctl`, `sysfs`, `vm`, `disk`, `net`, `modules`, and some of the
+`bootloader` settings such as `cmdline` and some `initrd` settings.  `tuned`
+is installed and enabled by default on Enterprise Linux and derivatives (RHEL
+and CentOS).  Using `tuned` allows managing all of the kernel parameters for
+performance tuning and security hardening, including bootloader parameters,
+all in one place (however it is no substitute for a general purpose
+bootloader/grub role).  `tuned` also persists settings upon reboot for those
+settings (such as `sysfs` settings) which would not otherwise be persisted (as
+opposed to `sysctl` which has an OS mechanism for applying settings at boot
+time).
+
+* `tuned` homepage - https://github.com/redhat-performance/tuned
+* `tuned` docs for EL7 - https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/performance_tuning_guide/chap-red_hat_enterprise_linux-performance_tuning_guide-tuned
+* `tuned` docs for EL8 - https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/monitoring_and_managing_system_status_and_performance/getting-started-with-tuned_monitoring-and-managing-system-status-and-performance
+
+## Requirements
+
+This role requires an operating system which has the `tuned` package and
+service.
+
+## Role Variables
+
+`kernel_settings_purge` - default `false` - If `true`, then the existing
+configuration will be completely wiped out and replaced with your
+`kernel_settings_parameters`.
+
+`kernel_settings_parameters` - default `{}` - This is a YAML `dict` of the
+settings you wish to apply.  The top level keywords such as `sysctl`, `sysfs`,
+etc. correspond to `tuned` plugins.  Beneath each top level keyword is a list
+or dict of settings to apply to that plugin. Each setting is a `dict` with
+allowed key names of `name`, `value`, `state`, and `previous`.  The settings
+are specified with a `name`, a `value`, and an optional `state` or `previous`
+which controls the behavior.  For example:
+```yaml
+kernel_settings_parameters:
+  sysctl:
+    - name: fs.epoll.max_user_watches
+      value: 785592
+    - name: fs.file-max
+      value: 379724
+  sysfs:
+    - name: /sys/kernel/debug/x86/pti_enabled
+      value: 0
+    - name: /sys/kernel/debug/x86/retp_enabled
+      value: 0
+    - name: /sys/kernel/debug/x86/ibrs_enabled
+      value: 0
+  vm:
+    - name: transparent_hugepages
+      value: madvise
+  bootloader:
+    - name: cmdline
```

Comment:
would it work to have it as a dict:
```yaml
bootloader:
  cmdline:
    - name: spectre_v2
      value: "off"
    - name: nopti
```
? It would simplify the syntax.

(For the settings above, there are name/value pairs, but those are different. For settings like sysctl, sysfs, the names are essentially arbitrary, constrained only by what the kernel supports. While `cmdline` under `bootloader` is, I suppose, a special entry. You will not have a list of multiple differently-named cmdlines. So it makes sense to make `cmdline` a dictionary key, not a value of `name`.)

Reply:
I think the disadvantage of making it inconsistent with the other settings, and the potential confusion caused, outweigh any advantage.